server/game/gameSystems/HealSystem.ts

```diff
@@ -37,9 +39,6 @@ export class HealSystem<TContext extends AbilityContext = AbilityContext> extend
         if (properties.isCost && (properties.amount === 0 || card.damage === 0)) {
             return false;
         }
-        if (card.hasRestriction(AbilityRestriction.BeHealed, context)) {
```

Comment:
Ah yeah okay I was thinking we might hit this issue. So we should keep this check here but make some changes to it.

Basically `canAffect()` for any system always should check if the card is a legal target for the effect and return false if not. By default it doesn't care if the effect will actually change game state or not, but there is a parameter to control that behavior which we need to update this code to respect.

If you look at `ExhaustSystem.canAffect()` you will see that it checks if the target is already exhausted but only if `properties.isCost` is true or the `mustChangeGameState` parameter set to some enum value other than "None". So basically by default it's only checking to see if the card is a legal target for exhaust, but if one of those two is set then it also checks that the effect must change game state. That's what we need here as well (and we should make the same change in the damage system now since we know we'll need it).